homeassistant/components/hue/hue_event.py

```diff
@@ -0,0 +1,101 @@
+"""Representation of a Hue remote firing events for button presses."""
+import logging
+
+from aiohue.sensors import TYPE_ZGP_SWITCH, TYPE_ZLL_ROTARY, TYPE_ZLL_SWITCH
+
+from homeassistant.const import CONF_EVENT, CONF_ID
+from homeassistant.core import callback
+from homeassistant.util import slugify
+
+from .sensor_device import GenericHueDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HUE_EVENT = "hue_event"
+CONF_LAST_UPDATED = "last_updated"
+CONF_UNIQUE_ID = "unique_id"
+
+EVENT_NAME_FORMAT = "{}"
+
+
+class HueEvent(GenericHueDevice):
+    """When you want signals instead of entities.
+
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, sensor, name, bridge, primary_sensor=None):
+        """Register callback that will be used for signals."""
+        super().__init__(sensor, name, bridge, primary_sensor)
+
+        self.event_id = slugify(self.sensor.name)
```

Comment:
Why would we slugify it ?

Reply:
It is also done in deCONZ.
I suppose it is because the sensor name may contain non-standard chars (like 'ñ'), spaces, uppercase and so on, so in order to be used as the event 'id', it is slugified like any entity id, just for coherency.